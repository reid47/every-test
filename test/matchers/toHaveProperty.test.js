import toHaveProperty from '../../src/matchers/toHaveProperty';

describe('toHaveProperty', () => {
  test('passing', () => {
    const el = document.createElement('div');
    el.something = 47;

    const result = toHaveProperty(el, 'something', 47);
    expect(result.pass).toBeTruthy();
    expect(result.message).toMatchSnapshot();
  });

  test('failing', () => {
    const el = document.createElement('div');
    el.something = 48;

    const result = toHaveProperty(el, 'something', 47);
    expect(result.pass).toBeFalsy();
    expect(result.message).toMatchSnapshot();

    const result2 = toHaveProperty(el, 'notPresent');
    expect(result2.pass).toBeFalsy();
    expect(result2.message).toMatchSnapshot();
  });
});
