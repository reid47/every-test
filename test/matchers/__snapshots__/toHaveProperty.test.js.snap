// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toHaveProperty failing 1`] = `"Expected element to have property \\"something\\" with value 47, but actual value was 48."`;

exports[`toHaveProperty failing 2`] = `"Expected element to have property \\"notPresent\\"."`;

exports[`toHaveProperty passing 1`] = `"Expected element not to have property \\"something\\" with value 47."`;
